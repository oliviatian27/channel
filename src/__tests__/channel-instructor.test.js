import React from 'react';
import { shallow } from 'enzyme';

import ChannelInstructor from '../components/channel-list/channel-instructor';


describe('ChannelInstructor tests', () => {

  it('renders ChannelInstructor component', () => {
    const Url = "apple";
    const Name = "pear"

    const wrapper = shallow(<ChannelInstructor instructorPhotoUrl={Url} instructorName={Name} />);

    // Expect the wrapper object to be defined
    expect(wrapper.find('.channel-instructor')).toBeDefined();
    expect(wrapper.find('.channel-instructor')).toMatchSnapshot();
  });

  it('renders InstructorPhoto and instructorName components and pass down the correct props', () => {
    const Url = "apple";
    const Name = "pear"

    const wrapper = shallow(<ChannelInstructor instructorPhotoUrl={Url} instructorName={Name} />);

    // Check if the prop in the child Component is correct
    expect(wrapper.find('InstructorPhoto').prop("instructorPhotoUrl")).toEqual(Url);
    expect(wrapper.find('InstructorName').prop("instructorName")).toEqual(Name);

    const newUrl = "peach";
    const newName = "grape"

   //update the props to check if the new prop passed down to the children component
   wrapper.setProps({ instructorPhotoUrl: newUrl,instructorName:newName });

   expect(wrapper.find('InstructorPhoto').prop("instructorPhotoUrl")).toEqual(newUrl);
   expect(wrapper.find('InstructorName').prop("instructorName")).toEqual(newName);
  });

})
