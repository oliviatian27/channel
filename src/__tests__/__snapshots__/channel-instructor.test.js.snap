// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelInstructor tests renders ChannelInstructor component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <ChannelInstructor
      instructorName="pear"
      instructorPhotoUrl="apple"
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InstructorPhoto
            instructorPhotoUrl="apple"
          />,
          <InstructorName
            instructorName="pear"
          />,
        ],
        "className": "channel-instructor",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "instructorPhotoUrl": "apple",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "instructorName": "pear",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <InstructorPhoto
              instructorPhotoUrl="apple"
            />,
            <InstructorName
              instructorName="pear"
            />,
          ],
          "className": "channel-instructor",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "instructorPhotoUrl": "apple",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "instructorName": "pear",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <InstructorPhoto
          instructorPhotoUrl="apple"
        />,
        <InstructorName
          instructorName="pear"
        />,
      ],
      "className": "channel-instructor",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "instructorPhotoUrl": "apple",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "instructorName": "pear",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InstructorPhoto
            instructorPhotoUrl="apple"
          />,
          <InstructorName
            instructorName="pear"
          />,
        ],
        "className": "channel-instructor",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "instructorPhotoUrl": "apple",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "instructorName": "pear",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InstructorPhoto
            instructorPhotoUrl="apple"
          />,
          <InstructorName
            instructorName="pear"
          />,
        ],
        "className": "channel-instructor",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "instructorPhotoUrl": "apple",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "instructorName": "pear",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
}
`;
